library/src/main/kotlin/com/a99/rxplaces/RxTextView.kt

```diff
@@ -0,0 +1,54 @@
+package com.a99.rxplaces
+
+import android.text.Editable
+import android.text.TextWatcher
+import android.widget.TextView
+import rx.Observable
+import rx.Subscriber
+import rx.android.MainThreadSubscription
+import rx.android.MainThreadSubscription.verifyMainThread
+
+class RxTextView {
+
+  companion object {
+    fun textChanges(textView: TextView): Observable<CharSequence> {
+      val observable = Observable.create<CharSequence> { subscriber ->
+        try {
+          verifyMainThread()
+
+          val watcher = createWatcher(subscriber)
+
+          textView.addTextChangedListener(watcher)
+
+          subscriber.onUnsubscribe { textView.addTextChangedListener(watcher) }
+        } catch (t: Throwable) {
+          subscriber.onError(t)
+        }
+      }
+
+      return observable.onBackpressureLatest()
+    }
+
+    private fun createWatcher(subscriber: Subscriber<in CharSequence>): TextWatcher {
+      return object : TextWatcher {
+        override fun afterTextChanged(s: Editable?) {}
+
+        override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+
+        override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+          if (!subscriber.isUnsubscribed) {
+            subscriber.onNext(s)
+          }
+        }
+      }
+    }
+
+    infix fun <T> Subscriber<T>.onUnsubscribe(function: () -> Unit) {
```

Comment:
Yeah, but I don't want to expose that function. That's why I've created as an inner extension function.